README.md

```diff
@@ -62,35 +62,42 @@ Select *Stash Pull Request Builder* then configure:
 
 ## Building the merge of Source Branch into Target Branch
 
-You may want Jenkins to build the merged PR (that is the merge of `sourceBranch` into `targetBranch`) to catch any issues resulting from this. To do this change the Branch Specifier from `origin/pr/${pullRequestId}/from` to `origin/pr/${pullRequestId}/merge`
+### Letting Stash do the merge
 
-If you are building the merged PR you probably want Jenkins to do a new build when the target branch changes. There is an advanced option in the build trigger, "Rebuild if destination branch changes?" which enables this.
+You may want Jenkins to build the merged PR (that is the merge of `sourceBranch` into `targetBranch`) to catch any issues resulting from this. To do this, change the Branch Specifier from `origin/pr/${pullRequestId}/from` to `origin/pr/${pullRequestId}/merge`
+
+If you are building the merged PR, you probably want Jenkins to do a new build when the target branch changes. There is an advanced option in the build trigger, "Rebuild if destination branch changes" which enables this.
 
 You probably also only want to build if the PR was mergeable and always without conflicts. There are advanced options in the build trigger for both of these. 
 
-**NOTE: *Always enable `Build only if Stash reports no conflicts` if using the merge RefSpec!*** This will make sure the lazy merge on stash has happened before the build is triggered.
+**NOTE: *Always enable `Build only if Stash reports no conflicts` if using the merge RefSpec!*** This will make sure the lazy merge on Stash has happened before the build is triggered.
+
+### Merging locally, variant 1
+
+If you don't want to use the lazy merged Stash PR RefSpec (described above), the other option is to do the merge locally as part of the build using the Jenkins git plugin (these only work for branches within the same repo):
+
+- In "Source Code Management" > "Git" > "Repositories" -> "Advanced" set "Refspec" to an empty value to enable the default behavior
+- Set the "Branch Specifier" to `*/${sourceBranch}`
+- In the "Source Code Management" > "Git" > "Additional Behaviors" section, click "Add" > "Merge Before Build"
+- In the "Name of repository" put "origin" (or, if not using default name, use your remote repository's name. Note: unlike in the main part of the Git Repository config, you cannot leave this item blank for "default".)
+- Set the "Branch to merge to" to `${targetBranch}`
```

Comment:
again, why not full refname?

Reply:
Will fix.

Reply:
It doesn't work. The "Merge Before Build" would prepend "origin/" or whatever is the name of the repository. "origin/refs/heads/master" worn't work.